auxiliary/scripts/pull-descriptions.py

```diff
@@ -0,0 +1,167 @@
+
+def main(args):
+    class_data = parse_class_data(args)
+    if args.generate_markdown:
+        generate_index(class_data)
+        generate_class_pages(class_data)
+    else:
+        for file_dict in class_data:
+            print(file_dict["name"])
+            for class_dict in file_dict["classes"]:
+                print(class_dict)
+            print()
+
+
+def generate_index(class_data):
+    """
+    Iterates over each of the source files indexed in `class_data`, and outputs
+    a single page of Markdown which is the documentation index.
+    """
+    contents = ""
+    for file_dict in class_data:
+        contents += "## %s\n" % file_dict["name"].title()
+        contents += "View source: [%s.py](https://github.com/ideoforms/isobar/tree/master/isobar/pattern/%s.py)\n\n" % (
+            file_dict["name"], file_dict["name"])
+        contents += "| Class | Function |\n"
+        contents += "|-|-|\n"
+        for class_dict in file_dict["classes"]:
+            contents += "| [%s](%s/%s.md) | %s |\n" % (class_dict["classname"],
+                                                       file_dict["name"],
+                                                       class_dict["classname"].lower(),
+                                                       class_dict["short_description"])
+        contents += "\n\n"
+    # Output to library.md
+    f = open("docs/patterns/library.md", "w")
+    f.write(contents)
+    f.close()
+
+def generate_class_pages(class_data):
+    for file_dict in class_data:
+        # Check if folder exists
+        if not os.path.exists("docs/patterns/%s" % file_dict['name'].lower()):
+            # Make the folder if not
+            os.mkdir("docs/patterns/%s" % file_dict['name'].lower())
+        for class_dict in file_dict['classes']:
+            # Format the file text into a list per line
+            class_content = []
+            class_content.append("# Pattern: %s" % class_dict['classname'])
+            class_content.append(class_dict['short_description'])
+            if (class_dict['long_description']):
+                class_content.append(class_dict['long_description'])
+            if (class_dict['arguments']):
+                class_content.append("## Arguments")
+                class_content.append(class_dict['arguments'])
+                print(class_dict['arguments'])
+            if (class_dict['example_output']):
+                class_content.append("## Example Output")
+                class_content.append("```py\n%s```" % class_dict['example_output'])
+            # Output to the proper file
+            fname = ("docs/patterns/%s/%s.md" % (file_dict['name'].lower(), class_dict['classname'].lower()))
+            f = open(fname, "w")
+            f.write("\n\n".join(class_content))
+            f.close()
+
+def parse_class_data(args):
+    # Get all pattern files, excluding __init__'s
+    pattern_files = list(sorted(glob.glob("isobar/pattern/[!_]*.py", recursive=True)))
+
+    allDicts = []
```

Comment:
To stick to Python [PEP8](https://peps.python.org/pep-0008/) guidelines, avoid camelCase and use underscored snake_case for variable names etc.

Reply:
Done for all camelCase variables, also added onto some other two-word  vars that would be helped with the readability and consistency